src/v/cluster/tx_types.h

```diff
@@ -0,0 +1,98 @@
+/**
```

Comment:
nit on naming, we already have rm_stm_types.h, perhaps use tm_stm_types in the same spirit (more than naming I think it suggests that the types here belong in the coordinator and `tx_types` is a little ambiguous).

Reply:
good point i will rename it